module/VuFind/src/VuFind/View/Helper/Root/JsConfigs.php

```diff
@@ -0,0 +1,151 @@
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Thomas Wagener <wagener@hebis.uni-frankfurt.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class JsConfigs extends AbstractHelper
+{
+    /**
+     * Config paths
+     *
+     * @var array
+     */
+    protected array $configPaths = [];
+
+    /**
+     * Constructor
+     *
+     * @param PluginManager $configLoader Config loader
+     */
+    public function __construct(
+        protected PluginManager $configLoader,
+    ) {
+    }
+
+    /**
+     * Add config paths to the internal array.
+     *
+     * @param array $new Paths to add
+     *
+     * @return void
+     */
+    public function addConfigPaths(array $new): void
+    {
+        $this->configPaths = array_merge_recursive($this->configPaths, $new);
+    }
+
+    /**
+     * Generate JSON from the internal arrays
+     *
+     * @return string
+     */
+    public function getJSON(): string
+    {
+        $res = [];
+        foreach ($this->configPaths as $configFile => $configPath) {
+            $config = $this->configLoader->get($configFile);
+            $res[$configFile] = $this->getConfigRecursive($configPath, $config);
+        }
+        $res = $this->filterNullAndEmpty($res);
+        if (empty($res)) {
+            return '{}';
+        }
+        return json_encode($res);
+    }
+
+    /**
+     * Get config recursive.
+     *
+     * @param string|array $configPaths Config paths
+     * @param ?Config      $config      Config
+     *
+     * @return array|string|null
+     */
+    protected function getConfigRecursive(string|array $configPaths, ?Config $config): array|string|null
+    {
+        if (!is_array($configPaths)) {
+            return $this->convertToArray($config?->$configPaths);
+        }
+        $res = [];
+        if (array_is_list($configPaths)) {
+            foreach ($configPaths as $configPath) {
+                $res[$configPath] = $this->convertToArray($config?->$configPath);
+            }
+        } else {
+            foreach ($configPaths as $configKey => $configPath) {
+                $res[$configKey] = $this->getConfigRecursive($configPath, $config?->$configKey);
+            }
+        }
+        return $this->filterNullAndEmpty($res);
+    }
+
+    /**
+     * Filter null and empty array.
+     *
+     * @param array $array Array
+     *
+     * @return array
+     */
+    protected function filterNullAndEmpty(array $array): array
+    {
+        return array_filter($array, function ($value) {
+            return $value !== null && (!is_array($value) || !empty($value));
+        });
+    }
+
+    /**
+     * Convert Config to array if necessary.
+     *
+     * @param null|string|Config $input Input
+     *
+     * @return null|string|array
+     */
+    protected function convertToArray(null|string|Config $input): null|string|array
```

Comment:
This might benefit from a different name, since it doesn't necessarily convert to an array -- it could return a string or null. Maybe `convertToPhpNative` would be better?

Reply:
I changed the name to convertToJsonSerializable since that is what that method is used for.

Reply:
Excellent name, thanks! :-)